### PR TITLE
[内容更新] src/docs/development/data-and-backend/json.md

### LINKED ISSUE #687: 'JSON 和序列化数据' page issue
Page URL: https://flutter.cn/docs/development/data-and-backend/json.html
Page source: https://github.com/cfug/flutter.cn/tree/master/src/docs/development/data-and-backend/json.md

Found a typo? You can fix it yourself by going to the page source and clicking the pencil icon. Or finish creating this issue.

Description of issue:

页面中仍包含部分英文没有翻译，页面底部的链接失效。
image

### PR DESCRIPTION
**Pull request by AlexV525**
Fix #687 .

### CODE REVIEW

**Review by cfug-dev — APPROVED** [review]

File: src/docs/development/data-and-backend/json.md [hunk 1]
```diff
@@ -242,15 +245,15 @@ become instantly more error-prone.
...
+类型安全、自动补全以及最重要的编译时异常。
 你的代码会立即变得更加容易出错。
 
 For example, whenever you access the `name` or `email` fields, you could quickly
 introduce a typo. A typo that the compiler doesn't know about since the
 JSON lives in a map structure.
 
 例如，当你获取 `name` 或者 `email` 字段，你可能很快引入一个笔误。
-然而编译器却无法知道 map 中有 JSON 笔误，编译器并不知道这个笔误。
+然而编译器却无法知道映射中瑟吉是否有 JSON 笔误。
```
Comment (AlexV525) [inline src/docs/development/data-and-backend/json.md:256]: Found a typo.
```suggestion
然而编译器却无法知道映射中是否有 JSON 笔误。
```

File: src/docs/development/data-and-backend/json.md [hunk 2]
```diff
@@ -65,9 +66,10 @@ For apps with several JSON models with more complexity,
 encoding by hand can quickly become tedious, repetitive, 
 and lend itself to many small errors.
 
-不同的项目复杂度不同，用例也不一样。对于较小的概念验证项目或者快速原型，
-使用代码生成器可能是过度的。对于具有很多更加复杂的 JSON 模型的 App，
-手动编码可能很快变得无聊，重复并且发生很多小错误。
+不同的项目复杂度不同，用例也不一样。
+对于较小的概念验证项目或者快速原型，使用代码生成器可能有些过于繁杂。
+对于具有很多更加复杂的 JSON 模型的 App，
```
Comment (AlexV525) [inline src/docs/development/data-and-backend/json.md:71]:
```suggestion
对于具有很多更加复杂的 JSON 模型的应用，
```

File: src/docs/development/data-and-backend/json.md [hunk 3]
```diff
@@ -337,8 +340,8 @@ works properly. In practice, the `User.fromJson()` and `User.toJson()`
 methods both need to have unit tests in place to verify correct behavior.
 
 通过这种方法，被调用的代码根本不需要担心序列化 JSON 数据的问题。
-然而，模型类仍然是必须的。在一个生产环境下的 App，
-你可能希望确保序列化数据能正确奏效。
+然而，你仍然需要模型类。
+你当然会希望序列化数据在一个生产环境的 App 里能奏效。
```
Comment (AlexV525) [inline src/docs/development/data-and-backend/json.md:344]:
```suggestion
你当然会希望序列化数据在一个生产环境的应用里能奏效。
```

File: src/docs/development/data-and-backend/json.md [hunk 4]
```diff
@@ -380,7 +386,7 @@ Although there are other libraries available, this guide uses
 [`json_serializable`][], an automated source code generator that
 generates the JSON serialization boilerplate for you.
 
-尽管有其它库可以使用，本指南使用了[`json_serializable`][]，
+尽管有其它库可以使用，但是本指南使用了[`json_serializable`][]，
```
Comment (AlexV525) [inline src/docs/development/data-and-backend/json.md:389]:
```suggestion
尽管有其它库可以使用，但是本指南使用了 [`json_serializable`][]，
```

File: src/docs/development/data-and-backend/json.md [hunk 5]
```diff
@@ -507,8 +512,8 @@ For example, if the API returns objects with _snake\_case_,
 and you want to use _lowerCamelCase_ in your models,
 you can use the `@JsonKey` annotation with a name parameter:
 
-如果需要，你可以很容易自定义命名策略。
-例如，如果 API 返回带有蛇形命名方式的对象，
+如果需要，你可以轻松自定义命名策略。
```
Comment (AlexV525) [inline src/docs/development/data-and-backend/json.md:515]:
```suggestion
如果需要，你可以很轻易地自定义命名策略。
```

File: src/docs/development/data-and-backend/json.md [hunk 6]
```diff
@@ -640,10 +654,12 @@ that the serialization works&mdash;it's now
...
 
-使用 `json_serializable`，在 `User` 类中你可以忘记手动序列化任意的 JSON 数据。
+在使用了 `json_serializable` 后，
+你可以立马忘掉 `User` 类中所有手动序列化的 JSON 数据。
 源代码生成器会创建一个名为 `user.g.dart` 的文件，
-它包含了所有必须的序列化数据逻辑。你不必再编写自动化测试来确保序列化数据奏效。
-现在由 _库来负责_ 确保序列化数据能正确地奏效。
+它包含了所有必须的序列化数据逻辑。
+你不必再编写自动化测试来确保序列化数据奏效。
+现在 **由库来负责** 确保序列化数据能正确地奏效。
```
Comment (AlexV525) [inline src/docs/development/data-and-backend/json.md:662]:
```suggestion
现在 **由库来负责** 确保序列化数据能正确地被转换。
```

**Pull request merged**